homeassistant/components/hisense_connectlife/climate.py

```diff
@@ -0,0 +1,777 @@
+            identifiers={(DOMAIN, device.device_id)},
+            name=device.name,
+            manufacturer="Hisense",
+            model=f"{device.type_name} ({device.feature_name})",
+        )
+        if device.feature_code == "19901":
+            self._attr_target_temperature_step = 0.5
+        # Get device parser to determine available modes and options
+        device_type = device.get_device_type()
+        if device_type:
+            try:
+                self._parser = coordinator.api_client.parsers.get(device.device_id)
+                self.static_data = coordinator.api_client.static_data.get(
+                    device.device_id
+                )
+                _LOGGER.debug(
+                    "Using parser for device type %s-%s:%s",
+                    device_type.type_code,
+                    device_type.feature_code,
+                    self._parser,
+                )
+                # Save device_type's type_code and feature_code for later use
+                self._current_type_code = device_type.type_code
+                self._current_feature_code = device_type.feature_code
+                # Set available modes based on device capabilities
+                self._setup_hvac_modes()
+                self._setup_fan_modes()
+                self._setup_swing_modes()
+            except Exception as err:  # noqa: BLE001  # noqa: BLE001
+                _LOGGER.error("Failed to get device parser: %s", err)
+                self._parser = None
+        else:
+            self._parser = None
+
+        # Only OFF mode if parser not available (don't assume device supports all modes)
+        if not hasattr(self, "_attr_hvac_modes"):
+            self._attr_hvac_modes = [HVACMode.OFF]
+
+        # Get target_temp attribute
+        target_temp_attr = (
+            self._parser.attributes.get(StatusKey.TARGET_TEMP) if self._parser else None
+        )
+
+        # Parse propertyValueList to get temperature range
+        def parse_temperature_range(property_value_list):
+            ranges = []
+            for item in property_value_list.split(","):
+                item = item.strip()
+                if "~" in item:
+                    lower, upper = map(int, item.split("~"))
+                    ranges.append((lower, upper))
+            return ranges
+
+        # Get parsed temperature range
+        if target_temp_attr and target_temp_attr.value_range:
+            temperature_ranges = parse_temperature_range(target_temp_attr.value_range)
+        else:
+            _LOGGER.warning(
+                "Target temperature attribute or value range not found, using default range."
+            )
+            temperature_ranges = [(MIN_TEMP, MAX_TEMP)]
+
+        # Get temp_type attribute
+        temp_type_attr = device.status.get(StatusKey.T_TEMP_TYPE)
+
+        # Select appropriate temperature range based on temp_type
+        if temp_type_attr != "1":
+            # Use first range (Celsius)
```

Comment:
You should be able to clean this up without all the if statements. Set it to min and max temp to default and unit to Celsius, if Fahrenheit adjust accordingly, if ranges, adjust accordingly.